IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m

```diff
@@ -105,16 +106,36 @@ - (BOOL)verifyResponse:(MSIDAADV1TokenResponse *)response
     {
         if (response.error)
         {
-            MSIDErrorCode errorCode = fromRefreshToken ? MSIDErrorServerRefreshTokenRejected : MSIDErrorServerOauth;
-
-            *error = MSIDCreateError(MSIDOAuthErrorDomain,
-                                     errorCode,
-                                     response.errorDescription,
-                                     response.error,
-                                     nil,
-                                     nil,
-                                     context.correlationId,
-                                     nil);
+            NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] initWithCapacity:1];
+            if (fromRefreshToken.homeAccountId)
+            {
+                [userInfo setObject:fromRefreshToken.homeAccountId forKey:MSIDUserIdKey];
+            }
+
+            if ([MSID_PROTECTION_POLICY_REQUIRED isEqualToString:response.additionalServerInfo[@"suberror"]])
```

Comment:
The service is currently sending this as "suberror", although the MSID constant is defined as "sub_error"